CMSgov/mint-app
# [EASI-2372] - Add Solution/Custom solution

# EASI-2372

## Changes and Description

- View for selecting additional, predefined solutions
- Option to select 'Other' solution
- View for inputing custom solution details
- View for input solution dates and status
- Storybooks for new operational need components
- Unit/cypress testing

Additional changes
- Added a custom testing util folder with `VerboseMockedProvider` for more verbose gql logging
- Components 
    - ImplementationStatus
    - HelpBox
    - NeedsStatus
    - SolutionCard

## /add-solution
![Screenshot 2022-11-30 at 11 07 08 AM](https://user-images.githubusercontent.com/95709965/204848802-1b36b222-0a11-4b12-83d4-79f77024a116.png)

## /add-custom-solution
![Screenshot 2022-11-30 at 11 08 13 AM](https://user-images.githubusercontent.com/95709965/204849073-4b59a348-b839-4229-96be-f48353e2f107.png)

## /solution-implementation-details
![Screenshot 2022-11-30 at 11 09 06 AM](https://user-images.githubusercontent.com/95709965/204849264-73cd91d2-5523-41a1-a3dd-144d0c230475.png)

## Storybook
![Screenshot 2022-11-30 at 11 14 21 AM](https://user-images.githubusercontent.com/95709965/204850519-b671bd85-dfab-4875-b659-264685a00b4b.png)



## How to test this change

- Seed database
- Answer a relevant operational need question/trigger
- Select a solution from opNeed table
- Click `Select a solution not listed`
- Add a predefined solution
- Repeat, but this time select `Other` form dropdown 
- Click `Add solution details` and fill out form
- Verify custom solution card
- Choose to update or remove the details then add the custom solution
- From the /add-solution view, click `Continue` and fill out solution implementation details
- Verify solutions are added to tracker table
- Try to break everything


## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
